File: samples/_shared/flutter/lib/src/sample_test_banner.dart
// Copyright 2026 Ivan Aliaga <ivan.aliaga@urp.edu.pe>
// SPDX-License-Identifier: Apache-2.0
//
// Thin top-of-screen banner used by the sample-test mode of every
// Dazzle Flutter sample. Tracks the phase exposed by `runSampleTest`
// (`preparing` / `running` / `completed` / `failed`) and colours the
// strip so a human watching the device can see at a glance whether the
// scripted conversation actually played out, independently of the
// JSON report the harness later validates.

import 'package:flutter/material.dart';

class SampleTestBanner extends StatelessWidget {
  const SampleTestBanner({super.key, required this.phase, this.detail});

  final String phase;
  final String? detail;

  @override
  Widget build(BuildContext context) {
    final (bg, text, icon) = _style(phase);
    return Container(
      width: double.infinity,
      color: bg,
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 10),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Icon(icon, size: 18, color: Colors.white),
          const SizedBox(width: 8),
          Flexible(
            child: Text(
              detail == null ? text : '$text  —  $detail',
              style: const TextStyle(
                color: Colors.white,
                fontWeight: FontWeight.w600,
              ),
              textAlign: TextAlign.center,
            ),
          ),
        ],
      ),
    );
  }

  (Color, String, IconData) _style(String phase) {
    switch (phase) {
      case 'preparing':
        return (
          const Color(0xFF1E3A8A),
          'DEV SMOKE · scripted (no real LLM) · preparing',
          Icons.hourglass_top,
        );
      case 'running':
        return (
          const Color(0xFF1E3A8A),
          'DEV SMOKE · scripted (no real LLM) · tap icon for Qwen',
          Icons.play_arrow,
        );
      case 'completed':
        return (
          const Color(0xFF166534),
          'DEV SMOKE · complete · app icon = real Qwen',
          Icons.check_circle,
        );
      case 'failed':
        return (
          const Color(0xFF7F1D1D),
          'DEV SMOKE · failed',
          Icons.error,
        );
      default:
        return (
          const Color(0xFF1E3A8A),
          phase,
          Icons.info,
        );
    }
  }
}

File: samples/_shared/android/SampleTestBanner.kt
// Copyright 2026 Ivan Aliaga <ivan.aliaga@urp.edu.pe>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
//
// Top-of-screen banner for the visible sample-test mode. Mirrors
// `SampleTestBanner` in the Flutter + RN samples so the cross-platform
// run looks consistent on device.

package dev.dazzle.samples.shared

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Error
import androidx.compose.material.icons.filled.HourglassTop
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

@Composable
fun SampleTestBanner(phase: String, detail: String?) {
    val (bg, label, icon) = when (phase) {
        "preparing" -> Triple(
            Color(0xFF1E3A8A),
            "DEV SMOKE · scripted (no real LLM) · preparing",
            Icons.Filled.HourglassTop,
        )
        "running" -> Triple(
            Color(0xFF1E3A8A),
            "DEV SMOKE · scripted (no real LLM) · tap icon for Qwen",
            Icons.Filled.PlayArrow,
        )
        "completed" -> Triple(
            Color(0xFF166534),
            "DEV SMOKE · complete · app icon = real Qwen",
            Icons.Filled.CheckCircle,
        )
        "failed" -> Triple(
            Color(0xFF7F1D1D),
            "DEV SMOKE · failed",
            Icons.Filled.Error,
        )
        else -> Triple(
            Color(0xFF1E3A8A),
            phase,
            Icons.Filled.PlayArrow,
        )
    }

    Row(
        Modifier
            .fillMaxWidth()
            .background(bg)
            .padding(horizontal = 12.dp, vertical = 10.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, contentDescription = null, tint = Color.White)
        Text(
            text = if (detail == null) label else "$label  —  $detail",
            color = Color.White,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier.padding(start = 8.dp),
        )
    }
}
